File: Session 1/code/network-java/src/main/java/com/compnet/practical1/Z1TCPServer.java
/*
 *
 */

package com.compnet.practical1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.net.ServerSocket;
import java.net.Socket;
import java.util.logging.Formatter;

/**
 *
 * @author
 */
public class Z1TCPServer {

    public static void main(String[] args) throws IOException {

        ServerSocket serverSocket = new ServerSocket(9999);

        // process clients one by one (not in parallel)
        while (true) {

            System.out.println("ACCEPTING CONNECTIONS");
            Socket socket = serverSocket.accept();
            System.out.println("GOT A CONNECTION");
            System.out.println("from ip: "+socket.getInetAddress());
            System.out.println("from port: "+socket.getPort());

            // Read two lines then ignore anything from this client
            InputStream inputStream = socket.getInputStream();
            String l = NetUtils.readLine(inputStream);
            System.out.println("RECEIVED FIRST: "+l);
            l = NetUtils.readLine(inputStream);
            System.out.println("RECEIVED SECOND: "+l);
            // ... ignore the rest

        }

    }

}

File: Session 1/code/network-python/Z1_tcp_server.py
import socket
import netutils

server_socket = socket.socket()
server_socket.bind(('localhost', 9999))
server_socket.listen()
print(server_socket)

# process clients one by one (not in parallel)
while True:
    print("Waiting for connection")
    socket, addr = server_socket.accept()
    print("Received connection", socket)
    # read two lines then ignore anything from this client...
    l = netutils.read_line(socket)
    print("RECEIVED FIRST:", l)
    l = netutils.read_line(socket)
    print("RECEIVED SECOND:", l)
    # ... ignore the rest
